code/modules/surgery/organs/lungs.dm

```diff
@@ -113,10 +160,389 @@
 	receiver.clear_alert(ALERT_NOT_ENOUGH_PLASMA)
 	receiver.clear_alert(ALERT_NOT_ENOUGH_N2O)
 
+/obj/item/organ/internal/lungs/Remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	// This is very "manuel" I realize, but it's useful to ensure cleanup for gases we're removing happens
+	// Avoids stuck alerts and such
+	var/static/datum/gas_mixture/immutable/dummy = new(BREATH_VOLUME)
+	for(var/gas_id in last_partial_pressures)
+		var/on_loss = breath_lost[gas_id]
+		if(!on_loss)
+			continue
+
+		call(src, on_loss)(organ_owner, dummy, last_partial_pressures[gas_id])
+	dummy.garbage_collect()
+
+/**
+ * Tells the lungs to pay attention to the passed in gas type
+ * We'll check for it when breathing, in a few possible ways
+ * Accepts 3 optional arguments:
+ *
+ * proc/while_present * Called while the gas is present in our breath. Return BREATH_LOST to call on_loss afterwards
+ * proc/on_loss * Called after we have lost a gas from our breath.
+ * proc/always * Always called. Best suited for breathing procs, like oxygen
+ *
+ * while_present and always get the same arguments (mob/living/carbon/breather, datum/gas_mixture/breath, pp, old_pp)
+ * on_loss is almost exactly the same, except it doesn't pass in a current partial pressure, since one isn't avalible
+ */
+/obj/item/organ/internal/lungs/proc/add_gas_reaction(gas_type, while_present = null, on_loss = null, always = null)
+	if(always)
+		breathe_always[gas_type] = always
+
+	if(!while_present && !on_loss)
```

Comment:
```suggestion
	if(isnull(while_present) && isnull(on_loss))
```

just based on the arg-defaults for this proc

Reply:
falsey args would be invalid too (isnull() is a spook and I don't like it but I'll do it if you really want me to)

Reply:
fucking mothbloaweddsjdjawds grumble grumble